autogen.sh

```diff
@@ -1,3 +1,6 @@
-#!/bin/sh
-test -d autoconf || mkdir autoconf
-autoreconf -i
+#!/bin/bash -e
```

Comment:
can't we stick with `sh` ?  there does not seem to be anything bash-specific in this script?

Reply:
As @ftynse stated, there is no need to use BASH instead of SHELL since there is nothing bash-specific. My bad, I just didn't think about other environments not supporting BASH.
However, I consider important to make the script fail whenever an error occurs on executing the internal commands because otherwise it gets complicated to handle errors when installing it in a larger pipeline.

Reply:
Reverted to `#!/bin/sh` and `set -e`. Removed unnecessary trailing whitespaces.